scripts/west_commands/runners/nrf_common.py

```diff
@@ -266,11 +277,25 @@ def program_hex(self):
         if self.family == 'NRF53_FAMILY':
             # nRF53 requires special treatment due to the extra coprocessor.
             self.program_hex_nrf53(erase_arg, qspi_erase_opt)
+        elif self.family == 'NRF54H_FAMILY':
+            self.program_hex_nrf54h()
         else:
             self.op_program(self.hex_, erase_arg, qspi_erase_opt, defer=True)
 
         self.flush(force=False)
 
+    def program_hex_nrf54h(self):
+        uicr_addrs = self.hex_get_uicrs()
+
+        if self.erase_all_uicrs:
+            for start_addr, _ in UICR_RANGES['NRF54H_FAMILY']:
+                self.exec_op('erasepage', defer=True, page=start_addr)
+        elif uicr_addrs:
+            for uicr_addr in uicr_addrs:
+                self.exec_op('erasepage', defer=True, page=uicr_addr)
```

Comment:
@57300 this will erase all UICRs in the hex file (we could now have app+app_uicr+rad_uicr)